src/Library/Plugin/Notice.php

```diff
@@ -0,0 +1,295 @@
+<?php
+/**
+ * BoldGrid Library Plugin Page Notice.
+ *
+ * @package Boldgrid\Plugin
+ *
+ * @since SINCEVERSION
+ * @author BoldGrid <wpb@boldgrid.com>
+ */
+
+namespace Boldgrid\Library\Library\Plugin;
+
+/**
+ * Notice class for Plugin\Page.
+ *
+ * This class is a specific Notice
+ * used by the Boldgrid\Library\Library\Plugin\Page 
+ * and Boldgrid\Library\Library\Plugin\Plugin classes.
+ *
+ * @since SINCEVERSION
+ */
+
+class Notice {
+
+    /**
+     * Notice ID
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $id;
+
+    /**
+     * Notice Page slug
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $pageSlug;
+
+    /**
+     * Notice Version
+     * 
+     * Version of plugin this notice was added on
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $version;
+
+    /**
+     * Plugin Object
+     * 
+     * Plugin Object this belongs to
+     * 
+     * @var Plugin
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $plugin;
+
+    /**
+     * Notice Is Unread
+     * 
+     * Specifies if the Notice is Unread or not
+     * 
+     * @var bool
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $isUnread;
+
+    /**
+     * Constructor 
+     * 
+     * @param Plugin $plugin Plugin instance that this Notice belongs to
+     * @param array $notice An array of Notice data from plugin config
+     * @since SINCEVERSION
+     * 
+     */
+    public function __construct( Plugin $plugin, array $notice ) {
+        if ( false !== $this->alreadyExists( $notice['id'] ) ) {
+            $originalNotice = $this->getNoticeFromOptions( $notice['id'] );
+            $this->id       = $originalNotice->id;
+            $this->pageSlug = $originalNotice->pageSlug;
+            $this->plugin   = $originalNotice->plugin;
+            if ( version_compare( $originalNotice->version, $notice['version'], '!=' ) ) {
+                $this->version   = $notice['version'];
+                $this->isUnread = true;
+            }
+            else {
+                $this->version = $originalNotice->getVersion();
+                $this->isUnread = $originalNotice->isUnread;
+            }
+            $this->updateNoticeOption( $this );
+        } else {
+            $this->id        = $notice['id'];
+            $this->pageSlug  = $notice['page']; 
+            $this->version   = $notice['version'];
+            $this->isUnread  = true;
+            $this->plugin    = $plugin;
+        }
+    }
+
+	/**
+	 * Get notice ID
+	 *
+	 * @return string
```

Comment:
Should be formatted like:

```
* @since SINCEVERSION
*
* @return string	 
```

Generally you want a space between `@since` and the `@param` and `@return` values.